flutter_appauth/android/src/main/java/io/crossingthestreams/flutterappauth/FlutterAppauthPlugin.java

```diff
@@ -479,7 +479,7 @@ private String getCauseFromException(Exception ex) {
 
     @Override
     public boolean onActivityResult(int requestCode, int resultCode, Intent intent) {
-        if (pendingOperation == null) {
+        if (pendingOperation == null || intent == null) {
```

Comment:
Thanks for the PR but I believe the check should be done further below. What's done here doesn't align with how onActivityResult works for Flutter plugins that override `onActivityResult` and what's done is results in a different approach to what's been done in the React Native AppAuth library. Flutter provides `onActivityResult` that returns a boolean so that plugins can indicate if they were the plugin that was meant to process the result (reference: https://api.flutter.dev/javadoc/io/flutter/plugin/common/PluginRegistry.ActivityResultListener.html). As you've put this check here, this results in the method returning `false` i.e. it would mean that plugin ends up reporting that it didn't handle the result. This would be incorrect and could end mean the intent is passed onto other plugins. Furthermore, it could cause the`Future` associated with the end session Dart API to not complete i.e. the app can hang on calling the end session API. This is because plugins should be using the [Result](https://api.flutter.dev/javadoc/io/flutter/plugin/common/MethodChannel.Result.html) object to report a success or error. Returning early here prevents this from happening. 

Where `intent` check should be done further is when the plugin is processing the result to do with an end session request. There's an existing "template" on this being done for authorisation checks at https://github.com/MaikuB/flutter_appauth/blob/f999c01d6e1811baf85b9a6b55ca2dfa89ce71f1/flutter_appauth/android/src/main/java/io/crossingthestreams/flutterappauth/FlutterAppauthPlugin.java#L486 so I believe correct solution is to mimic that and put the check at https://github.com/MaikuB/flutter_appauth/blob/f999c01d6e1811baf85b9a6b55ca2dfa89ce71f1/flutter_appauth/android/src/main/java/io/crossingthestreams/flutterappauth/FlutterAppauthPlugin.java#L496 where the plugin has confirmed that the result is associated with the request code for the end session request. Doing so would be similar to have the React Native plugin calls `promise.reject(...)` and brings consistency to how this plugin was already addressing this for authorisation requests

Reply:
Hmm I just realised that the code block for dealing with the end session request isn't returning `true`. Would you be able to make that change as well as part of the PR? If so, then that would be much appreciated

Reply:
yeah, just noticed as well. I made a fix for that. You can take a look

Reply:
LGTM, will merge it in once the checks are done. Thanks again :)